lib/ansible/modules/cloud/opennebula/one_vm.py

```diff
@@ -1430,10 +1430,10 @@ def main():
 
     # Fetch template
     template_id = None
-    if requested_template_id or requested_template_name:
+    if requested_template_id >= 0 or requested_template_name:
         template_id = get_template_id(module, one_client, requested_template_id, requested_template_name)
         if template_id is None:
-            if requested_template_id:
+            if requested_template_id >= 0:
```

Comment:
The initial idea here was to check whether `requested_template_id` is not None ie whether the parameter `template_id` was passed. Hence, you should go with `if requested_template_id is not None`...

Reply:
IMHO it behaves the same like 'is not None'